onetl/connection/db_connection/kafka.py

```diff
@@ -0,0 +1,223 @@
+
+    .. dropdown:: Version compatibility
+
+        * Apache Kafka versions: 0.10 or higher
+        * Spark versions: 2.3.x - 3.4.x
+
+    Parameters
+    ----------
+
+    addresses : list[str]
+        A list of broker addresses, for example ``[192.168.1.10:9092, 192.168.1.11:9092]``.
+        The list cannot be empty.
+
+    cluster : Cluster
+        Cluster name. Used for HWM and lineage. A cluster field cannot be empty.
+
+    user : str, default  ``None``
+        User, which have proper access to the Apache Kafka. For example: ``some_user``.
+
+    password : SecretStr, default  ``None``
+        Password for Kafka connection.
+
+    keytab : LocalPath, default  ``None``
+        A path to the keytab file. A keytab is a file containing pairs of Kerberos principals and encrypted keys that
+        are derived from the Kerberos password. You can use this file to log on to Kerberos without being prompted for
+        a password.
+
+    .. warning::
+
+        When creating a connector, when specifying `user` parameter, either `password` or `keytab` can be specified. Or
+        these parameters for anonymous connection are not specified at all.
+
+    Examples
+    --------
+
+    .. code:: python
+
+        kafka = Kafka(
+            addresses=["mybroker:9020", "anotherbroker:9020"],
+            cluster="mycluster",
+            user="me",
+            keytab="/etc/security/keytabs/me.keytab",
+            spark=spark,
+        )
+
+    .. code:: python
+
+        kafka = Kafka(
+            addresses=["mybroker:9020", "anotherbroker:9020"],
+            cluster="mycluster",
+            user="me",
+            password="pwd",
+            spark=spark,
+        )
+
+    """
+
+    def read_df(
+        self,
+        source: str,
+        columns: list[str] | None = None,
+        hint: Any | None = None,
+        where: Any | None = None,
+        df_schema: StructType | None = None,
+        start_from: Statement | None = None,
+        end_at: Statement | None = None,
+    ) -> DataFrame:
+        pass
+
+    def write_df(self, df: DataFrame, target: str) -> None:
+        pass
+
+    addresses: List[str]
+    cluster: Cluster
+    user: Optional[str] = None
+    password: Optional[SecretStr] = None
+    keytab: Optional[LocalPath] = None
+
+    @classmethod
+    def get_package_spark(
+        cls,
+        spark_version: str,
+        scala_version: str | None = None,
+    ) -> str:
+        if not scala_version:
+            scala_version = "2.11" if spark_version.startswith("2") else "2.12"
+        return f"org.apache.spark:spark-sql-kafka-0-10_{scala_version}:{spark_version}"
+
+    @validator("keytab")
+    def validate_keytab(cls, value):  # noqa: N805, U100
+        if not os.path.exists(value):
+            raise ValueError(
+                f"The file does not exists. File  properties:  {path_repr(value)} ",
```

Comment:
```suggestion
                f"File '{value}' is missing",
```
Какие properties могут быть у несуществующего файла?